peas/weather_web_data.py

```diff
@@ -0,0 +1,170 @@
+    leave, open or close the dome. They are also saved in a database so that previous
+    entries can be retrived.
+    """
+
+    def __init__(self, use_mongo=True):
+        super().__init__(use_mongo=use_mongo)
+
+        # Read configuration
+        self.web_data = self.config['weather']['web_service']
+
+        self.logger = logging.getLogger(self.web_data.get('name'))
+        self.logger.setLevel(logging.INFO)
+
+        self.max_age = TimeDelta(self.web_data.get('max_age', 60.), format='sec')
+
+        self.table_data = None
+
+
+    def capture(self, use_mongo=False, send_message=False, **kwargs):
+        self.logger.debug('Updating weather data')
+
+        data = {}
+
+        data['weather_data_from'] = self.web_data.get('name')
+        self.table_data = self.fetch_met_data()
+        col_names = self.web_data.get('column_names')
+        for name in col_names:
+            data[name] = self.table_data[name][0]
+
+        return super().capture(data)
+
+    def fetch_met_data(self):
+        try:
+            cache_age = Time.now() - self._met_data['Time (UTC)'][0] * 86400
+        except AttributeError:
+            cache_age = 1.382e10 * u.year
+
+        if cache_age > self.max_age:
+            # Download met data file
+            m = requests.get(self.web_data.get('link'))
+            # Remove the annoying " and newline between the date and time
+            met = m.text.replace('."\n',' ')
+            # Remove the " before the date
+            met = met.replace('" ', '')
+
+            # Parse the tab delimited met data into a Table
+            t = Table.read(met, format='ascii.no_header', delimiter='\t',
+                                names=self.web_data.get('column_names'))
+
+            # Convert time strings to Time
+            t['Time (UTC)'] = Time(t['Time (UTC)'], format='mixed_up_time')
+            # Change string format to ISO
+            t['Time (UTC)'].format = 'iso'
+            # Convert from AAT standard time to UTC
+            t['Time (UTC)'] = t['Time (UTC)'] - 10 * u.hour
+
+            col_names = self.web_data.get('column_names')
+            col_units = self.web_data.get('column_units')
+
+            if len(col_names) != len(col_units):
+                self.logger.debug('Number of columns does not match number of units given')
+
+            # Set units for items that have them
+            for name, unit in zip(col_names, col_units):
+                t[name].unit = unit
+
+            self._met_data = t
+
+        return self._met_data
+
+    def _get_cloud_safety(self, current_values):
+        sky_diff = self.weather_entries['sky-ambient']
+        sky_diff_u = self.weather_entries['sky-ambient uncertainty']
+
+        max_sky_diff = sky_diff + sky_diff_u
+        last_cloud = sky_diff
+
+        return super()._get_cloud_safety(max_sky_diff, last_cloud)
+
+    def _get_wind_safety(self, current_values):

```

Comment:
Modify to work with the statuses returned from `_get_status()`, not directly with the met data.

Reply:
Modified to use the results from `_get_status()`.